example/react/src/CasbinDemo.js

```diff
@@ -0,0 +1,32 @@
+import React, {Component} from 'react';
+import {Authorizer} from 'casbin.js';
+import permissions from './permissions.json'
+
+export default class CasbinDemo extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      user: props.user
+    }
+    this.auth = new Authorizer("manual");
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.setState({user: nextProps.user});
+  }
+
+  render() {
+    // set the permission when the user identity changes.
+    this.auth.setPermission(permissions[this.state.user]);
```

Comment:
consider network request, please move the code to `componentDidMount`.

Reply:
Nope, this is improper. `this.auth.setPermission()` here will not trigger any network request.

Maybe the react example doesn't properly designed and cannot clearly show the usage (and the mechanism) here. I will update this example later.

Reply:
render() is for display flow. We shouldn't "set" something in render(), right?